app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -0,0 +1,352 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show jsonEncode;
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:graphql/client.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+
+const String _labeledPullRequestsWithReviewsQuery = r'''
+query LabeledPullRequestsWithReviews($sOwner: String!, $sName: String!, $sLabelName: String!) {
+  repository(owner: $sOwner, name: $sName) {
+    labels(first: 1, query: $sLabelName) {
+      nodes {
+        id
+        pullRequests(first: 100, states: OPEN) {
+          nodes {
+            id
+            number
+            mergeable
+            commits(last:1) {
+              nodes {
+                commit {
+                  abbreviatedOid
+                  oid
+                  status {
+                    state
+                  }
+                }
+              }
+            }
+            reviews(first: 100, states: [CHANGES_REQUESTED, APPROVED]) {
```

Comment:
Can we do 
```
changeRequestedReviews: reviews(first: 1, states: [CHANGES_REQUESTED]) {
...
}
approvedReviews: reviews(first: 1, states: [APPROVED]) {
...
}
```
just in case that there are more than 100 reviews?

Reply:
100 is the max the API allows AFAIK.

I don't believe the queries you're proposing are valid, but if they are I'd be happy to change it.

Reply:
I was trying to use the [graphql aliases](https://graphql.org/learn/queries/#aliases) which seems to allow you to query the same thing multiple times with different conditions.

Reply:
Ahh very nice, this works.  Updated.